funcx_sdk/funcx/sdk/executor.py

```diff
@@ -47,62 +49,166 @@ def __repr__(self):
+                log.info("Attempting ws close")
+                await self.ws_handler.close()
+                log.info("Attempting ws re-connect")
+            else:
+                # clean exit
+                break
+
+    def shutdown(self):
+        if self.ws_handler is None:
+            return
+
+        self.ws_handler.closed_by_main_thread = True
+        ws = self.ws_handler._ws
+        if ws:
+            ws_close_future = asyncio.run_coroutine_threadsafe(
+                ws.close(), self.eventloop
+            )
+            ws_close_future.result()
+
+
 class FuncXExecutor(concurrent.futures.Executor):
     """Extends the concurrent.futures.Executor class to layer this interface
     over funcX. The executor returns future objects that are asynchronously
     updated with results by the WebSocketPollingTask using a websockets connection
     to the hosted funcx-websocket-service.
     """
 
+    funcx_client: FuncXClient
+    _function_future_map: t.Dict[str, Future]
+    _tasks: t.Dict[int, Future]
+    poller_thread: t.Optional[ExecutorPollerThread]
+
     def __init__(
         self,
-        funcx_client,
+        funcx_client: FuncXClient,
         label: str = "FuncXExecutor",
         batch_enabled: bool = False,
         batch_interval: float = 1.0,
         batch_size: int = 100,
     ):
-
         """
         Parameters
         ==========
 
         funcx_client : client object
             Instance of FuncXClient to be used by the executor
 
-        results_ws_uri : str
-            Web sockets URI for the results
-
         label : str
             Optional string label to name the executor.
             Default: 'FuncXExecutor'
         """
 
         self.funcx_client = funcx_client
-        # Disable throttling
-        self.funcx_client.throttling_enabled = False
 
-        self.results_ws_uri = self.funcx_client.results_ws_uri
         self.label = label
         self.batch_enabled = batch_enabled
         self.batch_interval = batch_interval
         self.batch_size = batch_size
-
-        self._tasks: t.Dict[str, Future] = {}
-        self._task_counter = 0
+        self.poller_thread = None
         self._function_registry: t.Dict[t.Any, str] = {}
-        self._function_future_map: t.Dict[str, Future] = {}
-        self.task_group_id = (
-            self.funcx_client.session_task_group_id
-        )  # we need to associate all batch launches with this id
+        self._tasks = {}
+        self._function_future_map = {}
 
+        self._reset_poller()
+        atexit.register(self.shutdown)
+
+    @property
+    def results_ws_uri(self) -> str:
+        return self.funcx_client.results_ws_uri
+
+    @property
+    def task_group_id(self) -> str:
+        return self.funcx_client.session_task_group_id
+
+    def _reset_poller(self):
+        self.shutdown()
+
+        self._task_counter = 0
+        to_cancel = itertools.chain(  # Satisfy the line-length linter
```

Comment:
This comment doesn't tell me much. What was done to satisfy `flake8`? This looks to me like a `black`ed invocation of a function, and nothing here strikes me as likely to be a problem for the line-length limit.

Reply:
Yes, the 88-character line-length.  Inlining directly into the `for` loop broke over multiple lines; could have gone with multiple `for` loops, but I opted for this.  Regardless, the majority of this thought-process was _not_ for the core logic, but how to make it most readable in the face of the 88-character length complaint.  The short synopsis of why _this_ approach as opposed to others I thought through is:
```python
 ... # Satisfy the line-length linter
```

Reply:
You can put an expression inlined in the for-loop or assign to a variable -- whatever you deem more readable -- and you'll satisfy the line-length requirement because that's taken care of by `black`.

So with that context, nothing _unusual_ is done here to satisfy the linters. A comment like this gets a closer read because its mere existence suggests that something special was done to keep things under 88 chars.